utils/test_reporter.go

```diff
@@ -0,0 +1,82 @@
+package utils
+
+import (
+	"fmt"
+	"os"
+	"testing"
+)
+
+/*
+This is a test reporter overriding testing.T's Fatalf to not runtime.Goexit() but rather os.Exit(1)
+See issue in: https://github.com/golang/mock/issues/145
+*/
+
+type TestReporter struct {
+	T *testing.T
+}
+
+func (r *TestReporter) Error(args ...interface{}) {
+	r.T.Error(args...)
+}
```

Comment:
Please change those too

```suggestion
type TestReporter struct {
	test *testing.T
}

func (reporter *TestReporter) Error(args ...interface{}) {
	reporter.T.Error(args...)
}
```

Reply:
This is a common pattern in go. Usually "`this`" receives a single letter or some generic name.